Slider/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -301,4 +320,76 @@ public static void SetSFXPitch(float value)
             s.source.pitch = s.pitch * value;
         }
     }
+
+    public static void EnqueueModifier(AudioModifier.ModifierType m)
+    {
+        if (_instance.modifiers.ContainsKey(m))
+            _instance.EnqueueModifier(_instance.modifiers[m]);
+        else
+            Debug.LogError($"Trying to access non-materialized modifier {m}");
+    }
+
+    private void EnqueueModifier(AudioModifier m)
+    {
+        foreach (var adj in m.adjustments)
+        {
+            var name = adj.parameter;
+            var val = adj.value;
+
+            if (parameterResponsibilityQueue.ContainsKey(name))
+                parameterResponsibilityQueue[name].Add(adj);
+            else {
+                parameterResponsibilityQueue.Add(name, new List<AudioModifier.AudioModifierProperty> { adj });
+                if (RuntimeManager.StudioSystem.getParameterByName(name, out float prev) == FMOD.RESULT.OK)
+                    parameterDefaults.TryAdd(name, prev);
+                else
+                    Debug.LogError($"Parameter {prev} is modified but does not actually exist");
+            }
+
+            SetGlobalParameter(name, val);
+        }
+    }
+
+    public static void DequeueModifier(AudioModifier.ModifierType m)
+    {
+        if (_instance.modifiers.ContainsKey(m))
+            _instance.DequeueModifier(_instance.modifiers[m]);
+        else
+            Debug.LogError($"Trying to access non-materialized modifier {m}");
+    }
+
+    private void DequeueModifier(AudioModifier m)
+    {
+        foreach (var adj in m.adjustments)
+        {
+            var name = adj.parameter;
+
+            if (parameterResponsibilityQueue.ContainsKey(name))
+            {
+                if (!parameterResponsibilityQueue[name].Remove(adj))
+                {
+                    Debug.LogError($"Modifier {m.name} is not actually in effect (parameter {name} has modifiers attached, but not {m.name})");
+                    continue;
+                }
+                if (parameterResponsibilityQueue[name].Count == 0)
```

Comment:
usually i like to add braces to if statements even if theyre one line bc i think it can prevent stupid bugs in the future where you have 2 lines that u think r in the if statement bc theyre indented, but its not bc theres no bracket

anyway esp if youre gonna put a comment for it (yay comment c:) u should prob put brackets around it for clarity